vsintegration/src/FSharp.LanguageService/FSharpProjectSite.fs

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.LanguageService
+
+open Microsoft.VisualStudio
+open Microsoft.VisualStudio.TextManager.Interop
+open Microsoft.VisualStudio.Shell
+open Microsoft.VisualStudio.Shell.Interop
+open Microsoft.VisualStudio.FSharp.LanguageService
+open Microsoft.FSharp.Compiler.SourceCodeServices
+open Microsoft.FSharp.Compiler.AbstractIL.Diagnostics 
+open Internal.Utilities.Collections
+open Internal.Utilities.Debug
+open System
+open System.IO
+open System.Diagnostics
+open System.Collections.Generic
+open Microsoft.CodeAnalysis
+open Microsoft.VisualStudio.LanguageServices.Implementation
+open Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+
+type internal FSharpProjectSite(hierarchy: IVsHierarchy, serviceProvider: System.IServiceProvider, visualStudioWorkspace: VisualStudioWorkspaceImpl, projectName: string) =
+    inherit AbstractProject(visualStudioWorkspace.ProjectTracker, null, projectName, hierarchy, "F#", serviceProvider, null, visualStudioWorkspace, null)
+
+    let mutable checkOptions : FSharpProjectOptions option = None
+
+    member internal this.CheckOptions = 
+            match checkOptions with 
+            | Some options -> options
+            | None -> failwith "Options haven't been computed yet."
+
+    member internal this.Initialize(hier: IVsHierarchy, site : IProjectSite) =
+        this.ProjectTracker.AddProject(this)
+
+        site.AdviseProjectSiteChanges(FSharpCommonConstants.FSharpLanguageServiceCallbackName,
+                                        new AdviseProjectSiteChanges(fun () -> this.OnProjectSettingsChanged(hier, site)))
+
+        site.AdviseProjectSiteClosed(FSharpCommonConstants.FSharpLanguageServiceCallbackName, 
+                                        new AdviseProjectSiteChanges(fun () -> this.Disconnect()))
+
+        // Add files and references.
+        site.SourceFilesOnDisk() |> 
```

Comment:
AFAIR `Seq.iter` is not marked as `inline` meaning that this call will allocate `FSharpFunc` object. Any reasons to prefer it to plain `for file in site.SourceFilesOnDisk() do this.Add.Document(hier, file)`?